packages/calypso-e2e/src/test/rest-api-client.test.ts

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// Disable the any check in order to mock private methods.
+
+import { describe, expect, jest, test } from '@jest/globals';
+import { SecretsManager } from '..';
+import { RestAPIClient } from '../rest-api-client';
+
+describe( 'RestAPIClient: private methods', function () {
+	test( 'Test: getBearerToken with valid credentials returns a valid bearer token', async function () {
+		const credentials = SecretsManager.secrets.testAccounts.defaultUser;
+		const restAPIClient = new RestAPIClient( {
+			username: credentials.username,
+			password: credentials.password,
+		} );
+		const bearerToken = await ( restAPIClient as any ).getBearerToken();
+
+		expect( bearerToken ).toBeDefined();
+		expect( typeof bearerToken ).toBe( 'string' );
+		expect( bearerToken.length ).toBe( 64 );
```

Comment:
Because these are actually making web requests over the network, I think these qualify as integration tests. Our unit tests I believe have the assumption that they should be able to run entirely in-memory on a machine without network.

With that, I think we probably need to either move these tests out of the unit test configs/build to an integration test config/build, or we should mock the network traffic. There are a couple of good libraries for doing that, [nock](https://www.npmjs.com/package/nock), which is already used in the monorepo, and [msw](https://www.npmjs.com/package/msw), which is not in the monorepo, but seems to have a nicer API.